csrc/id_model/id_model.cpp

```diff
@@ -946,30 +944,29 @@ Expr* findMatchingExpr(
       continue;
     }
 
+    // TODO: Update the comment
     // For the final loop promotion map, we want to find
     // promotions within the same loop groups. Note that that's
     // guaranteed when a new domain is replayed instead of reusing an
     // existing domain.
-    if (require_loop_mapped_promotion) {
```

Comment:
This is the main diff of this PR. It now always picks domains that are mapped in the loop graph.